sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/state/StateSchemaCompatibilityCheckerSuite.scala

```diff
@@ -313,22 +314,13 @@ class StateSchemaCompatibilityCheckerSuite extends SharedSparkSession {
     runSchemaChecker(dir, queryId, oldKeySchema, oldValueSchema,
       ignoreValueSchema = ignoreValueSchema)
 
-    val e = intercept[StateSchemaNotCompatible] {
+    val e = intercept[SparkUnsupportedOperationException] {
       runSchemaChecker(dir, queryId, newKeySchema, newValueSchema,
         ignoreValueSchema = ignoreValueSchema)
     }
-
-    assert(e.getMessage.contains("Provided schema doesn't match to the schema for existing state!"))
-    assert(e.getMessage.contains(newKeySchema.toString()))
-    assert(e.getMessage.contains(oldKeySchema.toString()))
-
-    if (ignoreValueSchema) {
-      assert(!e.getMessage.contains(newValueSchema.toString()))
-      assert(!e.getMessage.contains(oldValueSchema.toString()))
-    } else {
-      assert(e.getMessage.contains(newValueSchema.toString()))
-      assert(e.getMessage.contains(oldValueSchema.toString()))
-    }
+    assert(e.getErrorClass === "STATE_STORE_KEY_SCHEMA_NOT_COMPATIBLE" ||
+      e.getErrorClass === "STATE_STORE_VALUE_SCHEMA_NOT_COMPATIBLE")
+    assert(e.getMessage.contains("does not match existing"))
```

Comment:
Can we validate if the key schema or value schema error class will be thrown, based on the actual mismatch?

Reply:
Added an additional check if `ignoreValueSchema` is set to true